sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java

```diff
@@ -226,7 +226,14 @@ public static boolean isEnableMultipleWriteLocations(DatabaseAccount account) {
 
     public static <E extends CosmosClientException> void setRequestHeaders(CosmosClientException cosmosClientException,
                                                                            Map<String, String> requestHeaders) {
-        cosmosClientException.requestHeaders = requestHeaders;
+        Map<String, String> requestHeadersMap = new HashMap<>();
```

Comment:
This will create a new Map for the headers, and so you have to pay the additional cost of Map creation.
can we instead change the filtering when the message is logged:

You can have this in the CosmosClientException when the message is logged. this would help to pay the cost only if the message is logged.
```java
    @Override
    public String toString() {
        return getClass().getSimpleName() + "{" + "error=" + cosmosError + ", resourceAddress='"
                   + resourceAddress + '\'' + ", statusCode=" + statusCode + ", message=" + getMessage()
                   + ", causeInfo=" + causeInfo() + ", responseHeaders=" + filterSensitiveDate(responseHeaders) + ", requestHeaders="
                   + filterSensitiveDate(requestHeaders) + '}';
    }

    private List<Map.Entry<String, String>> filterSensitiveDate(Map<String, String> requestHeaders) {
        return requestHeaders.entrySet().stream().filter(entry -> !HttpConstants.HttpHeaders.AUTHORIZATION.equalsIgnoreCase(entry.getKey()))
            .collect(Collectors.toList());
    }
```

Reply:
Make sense !